include/boost/beast/websocket/impl/read.hpp

```diff
@@ -514,8 +514,12 @@ class stream<NextLayer, deflateSupported>::read_some_op
                     }
                 }
                 BOOST_ASSERT( ! impl.rd_done );
-                if( impl.rd_remain == 0 && impl.rd_fh.fin )
+                if (impl.rd_remain == 0 && impl.rd_fh.fin) {
                     impl.rd_done = true;
+                }
+                else {
+                    cont = true;
```

Comment:
`cont` is unrelated to the logic of the read operation itself; it determines whether the completion handler should be scheduled on an executor or executed directly. I suspect there might be an issue in your custom stream type, where an additional dispatch can make it work.